assignments/minishell/README.md

```diff
@@ -0,0 +1,174 @@
+Bash is both an interactive command language and a scripting language.
+It is used to interact with the file system, applications, operating system and more.
+
+For this assignment you will build a Bash-like shell with minimal functionalities like traversing the file system, running applications, redirecting their output or piping the output from one application into the input of another.
+The details of the functionalities that must be implemented will be further explained.
+
+## Shell Functionalities
+
+### Changing the Current Directory
+
+The shell will support a built-in command for navigating the file system, called `cd`.
+To implement this feature you will need to store the current directory path because the user can provide either relative or absolute paths as arguments to the `cd` command.
+
+The built-in `pwd` command will show the current directory path.
+
+Follow the following examples below to understand these functionalities.
+
+```sh
+> pwd
+/home/student
+> cd operating-systems/assignments/minishell
+> pwd
+/home/student/operating-systems/assignments/minishell
+> cd inexitent
+no such file or directory
+> cd /usr/lib
+> pwd
+/usr/lib
+```
+
+> **_NOTE:_** Using the `cd` command without any arguments or with more than one argument doesn't affect the current directory path.
+> Make sure this edge case is handled in a way that prevents crashes.
+
+### Closing the Shell
+
+Inputting either `quit` or `exit` should close the minishell.
+
+### Running an Application
+
+Suppose you have an executable named `sum` in the current directory.
+It takes arbitrarily many numbers as arguments and prints their sum to `stdout`.
+The following example shows how the minishell implemented by you should behave.
+
+```sh
+> ./sum 2 4 1
+7
+```
+
+If the executable is located at the `/home/student/sum` absolute path, the following example should also be valid.
+
+```sh
+> /home/student/sum 2 4 1
+7
+```
+
+Each application will run in a separate child process of the minishell created using [fork](https://man7.org/linux/man-pages/man2/fork.2.html).
+
+### Environment Variables
+
+Your shell will support using environment variables.
+The environment variables will be initially inherited from the `bash` process that started your minishell application.
+
+If an undefined variable is used, its value is the empty string: `""`.
+
+> **_NOTE:_** The following examples contain comments which don't need to be supported by the minishell.
+> They are present here only to give a better understanding of the minishell's functionalities.
+
+```sh
+> NAME="John Doe"                    # Will assign "John Doe" to NAME variable
+> AGE=27                             # Will assign 27 to AGE variable
+> ./identify $NAME $LOCATION $AGE    # Will translate to ./identify "John Doe" "" 27 because $LOCATION is not defined
+```
+
+A variable can be assigned to another variable.
+
+```sh
+> OLD_NAME=$NAME    # Will assign the value of the NAME variable to OLD_NAME
+```
+
```

Comment:
After we review the skeleton and solution, we need to specify if variables may be reassigned or not and how to handle this case.